pkg/goanalysis/linter.go

```diff
@@ -71,6 +70,24 @@ func (lnt *Linter) LoadMode() LoadMode {
 	return lnt.loadMode
 }
 
+func (lnt *Linter) WithDesc(desc string) *Linter {
+	lnt.desc = desc
+
+	return lnt
+}
+
+func (lnt *Linter) WithConfig(cfg map[string]any) *Linter {
+	if len(cfg) == 0 {
+		return lnt
+	}
```

Comment:
When do we faulty pass an empty config? Maybe it's not used but I'd expect this to just clear the config if any was set rather than a noop.

Reply:
Technically, the configuration is only set once, I don't see a use case where the configuration needs to be reset.

The current change split the map of map to 2 maps, and so before that, the 1st map was empty when the second map was empty.

So I check if the 2nd map is empty to avoid creating the 1st map if not needed.

It's not simple to explain.